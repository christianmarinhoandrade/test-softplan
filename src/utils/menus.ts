export const getMenu = (
) => {
  return [
    {
      title: 'teste',
      path: '/teste',
      name: 'agendamento'
    }
  ]
}
