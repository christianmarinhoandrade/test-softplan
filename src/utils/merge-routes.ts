import { RouteProps } from 'react-router-dom'
const routesContext = require.context('../pages', true, /.routes./)

export type RouteMergeProps = {
  name?: string
  path?: string
} & RouteProps

const routes: RouteMergeProps[] = []

routesContext.keys().forEach((route) => {
  routes.push(...routesContext(route).default)
})

export default routes
