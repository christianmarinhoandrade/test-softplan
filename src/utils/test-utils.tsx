import { ReactElement } from 'react'
import { render } from '@testing-library/react'

import { Provider as ProviderRedux } from 'react-redux'
import store from 'store'

import ThemeProvider from 'providers/theme'

const customRender = (ui: ReactElement) =>
  render(
    <ProviderRedux store={store}>
      <ThemeProvider>{ui}</ThemeProvider>
    </ProviderRedux>
  )

export * from '@testing-library/react'
export { customRender as render }
