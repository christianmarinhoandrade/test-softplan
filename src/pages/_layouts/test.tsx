import 'match-media-mock'
import { BrowserRouter } from 'react-router-dom'
import { render } from 'utils/test-utils'

import Layout from '.'

describe('<NotFound />', () => {
  it('should render layout', async () => {
    const { container } = render(
      <BrowserRouter>
        <Layout>
          <div />
        </Layout>
      </BrowserRouter>
    )
    expect(container.firstChild).toMatchSnapshot()
  })
})
