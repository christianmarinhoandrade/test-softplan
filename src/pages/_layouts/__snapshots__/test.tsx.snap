// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NotFound /> should render layout 1`] = `
.c0 {
  box-shadow: 0px 1px 1px rgba(0,0,0,0.1);
  position: fixed;
  width: 100%;
  z-index: 100;
  background: white;
  top: 0px;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 4rem;
}

.c2 {
  padding: 0.5rem 0;
}

.c2 img {
  width: 30px;
  height: 30px;
}

.c3 {
  background: #001965;
  padding: 3rem 1rem 0 1rem;
  height: 10rem;
  text-align: center;
  color: #3b97de;
}

.c3 img {
  width: 30px;
  height: 30px;
}

.c3 p {
  margin-top: 1rem;
}

<div
  class="App"
>
  <header
    class="c0"
  >
    <nav
      class="c1 container"
    >
      <a
        aria-label="Dogs - Home"
        class="c2"
        href="/pais/list"
      >
        <img
          alt="Logo"
          src="test-file-stub"
        />
      </a>
    </nav>
  </header>
  <main
    class="AppBody"
  >
    <section
      class="container"
    >
      <div />
    </section>
  </main>
  <footer
    class="c3"
  >
    <img
      alt="Logo"
      src="test-file-stub"
    />
    <p>
      Softplan | 
      2021
    </p>
  </footer>
</div>
`;
